### ISSUE TITLE
wide modifier in conditions?

### ISSUE THREAD

**Issue by wxsBSD** [VirusTotal/yara#522]
I'd like to be able to say something like this:

```pe.resources[0].name_string == "BAR" wide``` instead of the current way of doing things which is ```pe.resources[0].name_string == "B\x00A\x00R\x00"```. I've had a couple of people ask me why they couldn't match on "BAR" when the tool they were using was hiding the fact it was a wide character string. I'll happily make this change if it is wanted as I think it would be a quality of life improvement for users (and it also dovetails nicely with some patches I've been sitting on in the dotnet module).

**Comment by HostageBrain** [comment]
Why so strange syntax?
`pe.resources[0].name_string == "BAR" wide`
usually people used in C/C++ and maybe somewhere elsewhere such syntax:
`pe.resources[0].name_string == L"BAR"`
I think it's much more common and intuitive syntax

**Comment by wxsBSD** [comment]
Because YARA already uses the wide modifier in string definitions. I would prefer consistency here.

**Comment by plusvic** [comment]
Actually the solution I have in mind is something similar to @HostageBrain's proposal. Regular expressions already accept case insensitive modifier like this: `/foobar/i`, that could be extended to the wide modifier, for both regexps and strings. Something like: `/foobar/w` for regexps and `"foobar"w` for strings, or maybe `"foobar"/w`.

The same syntax could be use instead of the existing one in string declarations:

```
strings:
   $a = "foobar"/iw
   $a = /foobar/iw
```

I'm still considering it carefully because I don't want to have too many ways of doing the same thing.

**Comment by wxsBSD** [comment]
Yeah, I was aiming for consistency here too. I'll be happy to implement whatever you think is the right one.
